snuba/admin/clickhouse/common.py

```diff
@@ -0,0 +1,66 @@
+from typing import MutableMapping
+
+from snuba import settings
+from snuba.clickhouse.native import ClickhousePool
+from snuba.clusters.cluster import ClickhouseClientSettings, ClickhouseCluster
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage
+from snuba.utils.serializable_exception import SerializableException
+
+
+class InvalidNodeError(SerializableException):
+    pass
+
+
+class InvalidCustomQuery(SerializableException):
+    pass
+
+
+class InvalidStorageError(SerializableException):
+    pass
+
+
+def is_valid_node(host: str, port: int, cluster: ClickhouseCluster) -> bool:
+    nodes = cluster.get_local_nodes()
+    return any(node.host_name == host and node.port == port for node in nodes)
+
+
+CONNECTIONS: MutableMapping[StorageKey, ClickhousePool] = {}
+
+
+def get_clickhouse_connection(
```

Comment:
```suggestion
def get_clickhouse_readonly_connection(
```
It makes it clear that noone can do do inserts/alters when they use this API